### ISSUE TITLE
Limit size of exception details in message field of event.

### ISSUE THREAD

**Issue by GrahamDumpleton** [zalando-incubator/kopf#87]
When an exception occurs, the full exception traceback is being added to the `message` field of an event. When this is posted to Kube, it result in an error as it can be too big to store.

```
HTTP response headers: HTTPHeaderDict({'Cache-Control': 'no-store', 'Content-Type': 'application/json', 'Dat
e': 'Thu, 30 May 2019 02:05:15 GMT', 'Content-Length': '392'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Event \"k
opf-event-5sqgc\" is invalid: message: Invalid value: \"\": can have at most 1024 characters","reason":"Inva
lid","details":{"name":"kopf-event-5sqgc","kind":"Event","causes":[{"reason":"FieldValueInvalid","message":"
Invalid value: \"\": can have at most 1024 characters","field":"message"}]},"code":422}
```

Should perhaps restrict what is used to lowest stack frame and exception description.

**Comment by nolar** [comment]
@GrahamDumpleton Can you please tell which exception is this? Or share a code snippet to reproduce it. 

Technically, a string representation of an exception is used (`f"{exc}"` in `kopf.structs.status. store_failure()`), which is usually just its type+message, not the stacktrace.

**Comment by nolar** [comment]
Fixed in `kopf==0.14`: https://github.com/zalando-incubator/kopf/releases/tag/0.14

Specifically:

* The event messages are cut to 1024 chars, keeping the start & end of the original string.
* Any errors in event posting are logged, but do not break the operator anymore (as events are auxiliary by their nature).
